python/friesian/example/dien/pytorch/yarn_dien_orca.py

```diff
@@ -0,0 +1,213 @@
+import torch.nn as nn
+import numpy as np
+from pyspark.sql.functions import lit, rand, col
+import torch.optim as optim
+from bigdl.orca import init_orca_context, stop_orca_context
+from bigdl.orca.learn.pytorch import Estimator
+from bigdl.orca.learn.metrics import Accuracy, BinaryCrossEntropy
+from bigdl.orca import OrcaContext
+from torch.nn import functional as F
+from bigdl.orca.learn.trigger import EveryEpoch
+
+import argparse
+import math
+import time
+
+from bigdl.orca import init_orca_context, stop_orca_context, OrcaContext
+from bigdl.orca.learn.pytorch import Estimator
+from bigdl.friesian.feature import FeatureTable
+from pyspark.sql.window import Window
+from pyspark.sql.functions import desc, rank, col
+
+input_meta = '/dien/meta_Books.csv'
+input_transaction = '/dien/reviews_Books.json'
+output = '/dien/data/'
+
+n_uid = 8026325
+n_mid = 2370586
+n_cat = 2752
+
+EMBEDDING_DIM = 18
+HIDDEN_SIZE = 18 * 2
+ATTENTION_SIZE = 18 * 2
+MAX_HIST_LEN = 100
+SEED = 3
+best_auc = 0.0
+hash_flag = False
+use_neg = False
+
+# build input feature table
+feature_columns = [SparseFeat('user', n_uid, embedding_dim=EMBEDDING_DIM, use_hash=hash_flag),
+                    SparseFeat('item_id', n_mid, embedding_dim=EMBEDDING_DIM, use_hash=hash_flag),
+                    SparseFeat('cate_id', n_cat, embedding_dim=EMBEDDING_DIM, use_hash=hash_flag)]
+feature_columns += [
+    VarLenSparseFeat(SparseFeat('hist_item_id', vocabulary_size=n_mid, embedding_dim=EMBEDDING_DIM, embedding_name='item_id'),
+                        maxlen=MAX_HIST_LEN, length_name="seq_length"),
+    VarLenSparseFeat(SparseFeat('hist_cate_id', vocabulary_size=n_cat, embedding_dim=EMBEDDING_DIM, embedding_name='cate_id'),
+                        maxlen=MAX_HIST_LEN,
+                        length_name="seq_length")]
+
+behavior_feature_list = ["item_id", "cate_id"]
+
+
+def model_creator(config):
+    model = DIEN(feature_columns, behavior_feature_list, dnn_hidden_units=[4, 4, 4], dnn_dropout=0.4, gru_type="AUGRU",
+                 use_negsampling=False, device='cpu', seed=2, att_hidden_units=(64, 16), att_activation="relu", init_std=0.1)#, modelpath=root_path)
+    return model
+
+def optim_creator(model, config):
+    optimizer = optim.Adam(model.parameters(),
+                          lr=config.get("lr", 0.001))
+    return optimizer
+
+if __name__ == '__main__':
+    OrcaContext.log_output = True  
+    executor_cores = 28
+    num_executor = 2
+    executor_memory = "30g"
+    driver_cores = 4
+    driver_memory = "36g"
+    conf = {"spark.executor.memoryOverhead": "130g",
+            "spark.network.timeout": "10000000",
+            "spark.sql.broadcastTimeout": "7200",
+            "spark.sql.shuffle.partitions": "2000",
+            "spark.locality.wait": "0s",
+            "spark.sql.crossJoin.enabled": "true",
+            "spark.task.cpus": "1",
+            "spark.executor.heartbeatInterval": "200s",
+            "spark.driver.maxResultSize": "40G",
```

Comment:
refer to other examples for this, don't directly copy from notebook version.
OrcaContext.log_output = True  is needed for jupyter only.